python/cudf/cudf/core/index.py

```diff
@@ -767,6 +774,86 @@ def isin(self, values):
 
         return result
 
+    def replace(self, to_replace, replacement):
+        """
+        Replace values given in *to_replace* with *replacement*.
+
+        Parameters
+        ----------
+        to_replace : numeric, str, list-like or dict
+            Value(s) to replace.
+
+            * numeric or str:
+
+                - values equal to *to_replace* will be replaced
+                  with *replacement*
+
+            * list of numeric or str:
+
+                - If *replacement* is also list-like,
+                  *to_replace* and *replacement* must be of same length.
+
+            * dict:
+
+                - Dicts can be used to replace different values in different
+                  columns. For example, `{'a': 1, 'z': 2}` specifies that the
+                  value 1 in column `a` and the value 2 in column `z` should be
+                  replaced with replacement*.
+        replacement : numeric, str, list-like, or dict
+            Value(s) to replace `to_replace` with. If a dict is provided, then
+            its keys must match the keys in *to_replace*, and corresponding
+            values must be compatible (e.g., if they are lists, then they must
+            match in length).
+
+        Returns
+        -------
+        result : Index, StringIndex
+            Index after replacement. Replace values given in *to_replace* with 
+            *replacement*.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> idx = cudf.core.index.StringIndex(["a","b","c"])
+        >>> idx_replaced = idx.replace(to_replace=["a", "b"], replacement=[1, 2])
+        >>> idx_replaced
+        StringIndex(['1' '2' 'c'], dtype='object')
+
+        Notes
+        -----
+        Parameters that are currently not supported are: `limit`, `regex`,
+        `method`
+        
+        If any item from source or dest is a string, normalize all items 
+        to be strings.
+        """
+        if any(
+            map(
+                lambda x: is_string_dtype(as_column(x)),
+                [self._values, to_replace, replacement],
+            )
+        ):
+
+            str_normalize_dt = (
+                lambda x: as_column(x)
+                if is_string_dtype(as_column(x))
+                else as_column(list(map(str, x)))
+            )
+
+            return as_index(
+                libcudf_replace(
+                    str_normalize_dt(self._values),
+                    str_normalize_dt(to_replace),
+                    str_normalize_dt(replacement),
+                )
+            )
+        else:
+            return as_index(
+                libcudf_replace(
+                    self._values, as_column(to_replace), as_column(replacement)
+                )
+            )
```

Comment:
Do we need to redefine this or can we just inherit from `Frame.replace`?

Reply:
That was my initial intention but I ran into issues with the specifics of index such as when building result from_table, the object in self.index for an index doesn't have an index. Sure, can check and drop that, but then the `column.find_and_replace` had problems with type handling for indexes. Because index seems to be a unique case I felt this was cleaner. If you would like I can roll the logic into `frame.replace`.

Reply:
I think we have the pattern elsewhere for handling DataFrame vs Series vs Index in a general way in Frame. That seems like much better maintenance burden to handle.

Reply:
OK. I will do that. I'll look for an existing example.

Reply:
I think this is a good example: https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/core/frame.py#L1174

In the case of an Index / MultiIndex `self._index` will be `None`.